### PR TITLE
Adds Google Analytic tracking of shares on the congrats page.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Allows for metrics by social platform (e.g., Facebook and Twitter) and tutorial (e.g., Angry Birds and Frozen). Tested locally via the Chrome Google Analytics Debugger plugin.

congratsshareprpic

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Though I think it would be nice to have sharing metrics sooner rather than later, if appropriate, submission can certainly wait.

**Comment by alicesteinglass** [comment]
I agree with Asher that it would be nice to have this before tomorrow so we can see if people are using share when we release the tutorial. However, Phil, I don't want to put this release at risk. If you are worried about taking this now, let's wait.

File: pegasus/sites.v3/code.org/views/congrats_share.haml [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
+    trackEvent(social_platform, "share_hoc_certificate", tutorial);
+  }
+
 #congrats-share-buttons
   %h2{:style=>'margin-top: 0; margin-bottom: 0;'}=I18n.t :share_achievement
 
   .share-buttons.clear
     %div#share-fb{:style=>"display:inline-block; margin-bottom: 10px; margin-right: 20px"}
-      %a.window-popup{:href=>"https://www.facebook.com/sharer/sharer.php?#{facebook.to_query}"}<
+      %a.window-popup{:href=>"https://www.facebook.com/sharer/sharer.php?#{facebook.to_query}", onclick: "trackSocialEvent('facebook'); return false;"}<
```
Comment (philbogle) [inline pegasus/sites.v3/code.org/views/congrats_share.haml]: I believe returning false here will cause the browser not to navigate to the link, so I think the sharing won't actually happen.

I tried a small jsfilddle to confirm this.

**Comment by philbogle** [comment]
I believe we should not take this PR before Monday; there's not enough cycles to test it adequately and the non-zero risk of breaking social sharing outweights the benefit.  We can see social sharing indirectly through actrivity on the site.

File: pegasus/sites.v3/code.org/views/congrats_share.haml (hunk 1 above)
Comment (ashercodeorg) [inline pegasus/sites.v3/code.org/views/congrats_share.haml]: Removed the return false statements. Note that, running locally [before the change], I was seeing the FB (Twitter) login modal and the CSS flip (what icons are visible). So I'm not sure what behavior you are seeing.

**Comment by philbogle** [comment]
The spec for onclick says that if return false the href won't be followed.  (If it is navigating to the destination, this might be a sign that your handler is throwing an exception.)

See for example:
http://www.javascripttoolbox.com/bestpractices/
"The javascript code that runs within the onclick handler must return true or false (or an expression than evalues to true or false) back to the tag itself - if it returns true, then the HREF of the anchor will be followed like a normal link. If it returns false, then the HREF will be ignored. This is why "return false;" is often included at the end of the code within an onclick handler."

**Comment by ashercodeorg** [comment]
@philbogle Is this safe for submission now? Does it have your LGTM?

File: pegasus/sites.v3/code.org/views/congrats_share.haml [hunk 2]
```diff
@@ -1,9 +1,31 @@
+:javascript
+  // Triggers the registration of a Google Analytics "share_hoc_certificate" event.
+  // Args:
+  //   social_platform: The social platform the HOC certificate is being shared on. 
+  function trackSocialEvent(social_platform) {
+    // Determine the tutorial being shared via the document referrer.
+    if (!document.referrer) {
```
Comment (philbogle) [inline pegasus/sites.v3/code.org/views/congrats_share.haml:7]: We might want to think about a more robust/maintainable way of doing this longer term-- we shouldn't have to parse it out of the referer. This is fine for now however.

**Comment by philbogle** [comment]
LGTM, see comment regarding a possible future improvement.

**Pull request merged**